### PR TITLE
MGDOBR-674: Plug "Processor detail" page to APIs

### PR DESCRIPTION
**Pull request by kelvah**
https://issues.redhat.com/browse/MGDOBR-674

Steps to test it:

- `npm run start:mocked`
- open example processors urls: [sink processor](http://localhost:8080/instance/3543edaa-1851-4ad7-96be-ebde7d20d717/processor/a72fb8e7-162b-4ae8-9672-f9f5b86fb3d7), [source processor](http://localhost:8080/instance/3543edaa-1851-4ad7-96be-ebde7d20d717/processor/f8f34af4-caed-11ec-9d64-0242ac120002)

### CODE REVIEW

File: src/app/Processor/ProcessorDetail/ProcessorDetail.tsx [hunk 1]
```diff
@@ -134,18 +144,32 @@ const ProcessorDetail = (props: ProcessorDetailProps): JSX.Element => {
...
-                {t("processor.transformationTemplate")}
-              </Text>
-            </TextContent>
-          </StackItem>
-          <StackItem>
-            <CodeBlock className="processor-detail__transformation-template">
-              <CodeBlockCode>{processor.transformationTemplate}</CodeBlockCode>
-            </CodeBlock>
-          </StackItem>
+          {processor.transformationTemplate ? (
```
[the remark this answers is not in the document]
Comment (kelvah) [reply src/app/Processor/ProcessorDetail/ProcessorDetail.tsx]: Good catch. I've read it wasn't supported yet but I totally forgot to remove the related section. Done.

**Review by manstis — APPROVED** [review]
Approving, but could you please share why you first added but now reverted the async functions in our API hooks? Was `React` erm, not reacting to the asynchronous data changes or something?

**Comment by kelvah** [comment]
> Approving, but could you please share why you first added but now reverted the async functions in our API hooks? Was `React` erm, not reacting to the asynchronous data changes or something?

`await` lets you manage async code without using callbacks (like `.then()` etc.). In this case we are actually managing the bridgeApi call with callbacks, so the async/await combination was there to no use.

By the way they were already there, I only removed them.

File: src/app/Processor/ProcessorDetail/ProcessorDetailSkeleton.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  StackItem,
+} from "@patternfly/react-core";
+
+const ProcessorDetailSkeleton = (): JSX.Element => {
+  return (
+    <>
+      <PageSection variant={PageSectionVariants.light}>
+        <Stack hasGutter={true}>
+          <StackItem>
+            <Skeleton width="150px" screenreaderText="Loading processor type" />
```
Comment (vpellegrino) [inline src/app/Processor/ProcessorDetail/ProcessorDetailSkeleton.tsx:20]: Maybe this text should be put under i18n

**Review by vpellegrino — APPROVED** [review]
PR looks good to me. Just a minor comment.
Thank you @kelvah !

**Comment by sjamboro** [comment]
@kelvah in case that you want to fix the `SDK<->Openapi` check. Just rebase on the top of main.

File: locales/en/openbridge.json [hunk 3]
```diff
@@ -53,6 +53,10 @@
     "deleteFilter": "Delete filter",
     "filters": "Filters",
     "generalInformation": "General information",
+    "loadingProcessor": "Loading processor",
+    "loadingProcessorType": "Loading processor type",
```
Comment (jomarko) [inline locales/en/openbridge.json:57]: What are the steps user should be able to see this message please? Once I clicked on a processor in the instance detail page, the screen turn to grey (it was clear to me loading is happening), but no message about it appears. Is that expected?

**Review by jomarko — COMMENTED** [review]
@kelvah thank you for the PR.

One comment inline. Another is related to click on 'Processor two' on the 'Instance one' detail page. It was leading to nowhere. I changed it for 'Processor three' See my PR https://github.com/kelvah/sandbox-ui/pull/4 and share your opinion please.

File: locales/en/openbridge.json (hunk 3 above)
Comment (kelvah) [reply locales/en/openbridge.json:57]: Yes, it is the expected behavior. 

They are screen reader descriptions passed to PF Skeleton component (like [here](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox-ui/blob/0a82751c5ab0547dbaa065660659d1123b590683/src/app/components/PageHeaderSkeleton/PageHeaderSkeleton.tsx#L44)), so they will be read  only by screen reader users.

**Review by jomarko — APPROVED** [review]
Thank you @kelvah

**Comment by kelvah** [comment]
> @kelvah thank you for the PR.
> 
> One comment inline. Another is related to click on 'Processor two' on the 'Instance one' detail page. It was leading to nowhere. I changed it for 'Processor three' See my PR [kelvah#4](https://github.com/kelvah/sandbox-ui/pull/4) and share your opinion please.

@jomarko I've merged your PR even if the hard-coded processors list is short lived. It will be removed soon as Valentino started working on [MGDOBR-673](https://issues.redhat.com/browse/MGDOBR-673).

**Pull request merged**